src/battle_main.c

```diff
@@ -3720,6 +3720,10 @@ static void TryDoEventsBeforeFirstTurn(void)
         gBattlerAttacker = gBattlerByTurnOrder[gBattleStruct->switchInAbilitiesCounter++];
         if (AbilityBattleEffects(ABILITYEFFECT_ON_SWITCHIN, gBattlerAttacker, 0, 0, 0) != 0)
             return;
+        if (gBattleWeather & B_WEATHER_ANY && WEATHER_HAS_EFFECT && AbilityBattleEffects(ABILITYEFFECT_ON_WEATHER, gActiveBattler, 0, 0, 0) != 0)
```

Comment:
sure they can. if another mon has electric surge for example, it will call `tryactivateterrainchangeabilities` which would trigger mimicry, for example

Reply:
I'm talking about instances in which a Pokémon with those abiliies is sent out on the very first turn and the conditions match.

Feel free to test it out yourself by cloning the branch I'm PR'ing here if you don't believe me, though I can record a video too.

I assigned Mimicry as Charmander's ability, removed the check for `ABILITYEFFECT_ON_TERRAIN` here and entered a battle while the Rain Thunderstorm weather was active in the overworld.

What happened? Mimicry wasn't activated when the battle began, after the Electric Terrain was invoked.

The exact same thing happened with Protosynthesis. I assigned it as Charmander's ability, removed the check that uses `ABILITYEFFECT_ON_WEATHER` and entered a battle while the `WEATHER_DROUGHT` was in effect.

What happened? Charmander couldn't activate Protosynthesis.